src/aap_eda/tasks/orchestrator.py

```diff
@@ -184,4 +223,113 @@ def monitor_rulebook_processes() -> None:
         else:
             id = process.event_stream_id
         job_id = _manage_process_job_id(process_parent_type, id)
-        unique_enqueue("activation", job_id, _manage, process_parent_type, id)
+        queue = _get_queue_name(process_parent_type, id)
+        unique_enqueue(queue, job_id, _MANAGE_TASK, process_parent_type, id)
+
+
+def monitor_engine_events() -> None:
+    """Monitor container engine events.
+
+    Started by the activation worker once, before it kicks on activation
+    It enqueues a monitor task for each activation when it ends
+    """
+    LOGGER.info("Monitor engine events has started")
+    container_engine = new_container_engine(1, ProcessParentType.ACTIVATION)
+    for event in container_engine.monitor_events():
+        # monitor running instances
+        LOGGER.info(f"Container engine event {event}")
+        if event.reason in ["died", "Completed", "BackoffLimitExceeded"]:
```

Comment:
I think this should be a enum defined from the Event dataclass which acts as a interface for the engine, otherwise you have to deal with free strings that comes from different engines.